File: VB/CodeExamples/VbaMacrosActions.vb
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native
Imports System
Imports System.Collections.Generic

Namespace RichEditDocumentServerAPIExample.CodeExamples

    Public Module VbaMacrosActions

        Public ObtainVbaMacrosAction As Action(Of DevExpress.XtraRichEdit.RichEditDocumentServer) = AddressOf ObtainVbaMacros

        Public ClearVbaModulesAction As Action(Of DevExpress.XtraRichEdit.RichEditDocumentServer) = AddressOf ClearVbaModules

        Private Sub ObtainVbaMacros(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#ObtainVbaMacros"
            Dim document As DevExpress.XtraRichEdit.API.Native.Document = wordProcessor.Document
            document.LoadDocument("Documents\Grimm.docx")
            If document.VbaProject.Modules.Count > 0 Then
                For Each [module] As DevExpress.XtraRichEdit.API.Native.VbaModule In document.VbaProject.Modules
                    document.AppendText(Global.Microsoft.VisualBasic.Constants.vbCrLf & " · " & [module].Name)
                Next
            End If
#End Region  ' #ObtainVbaMacros
        End Sub

        Private Sub ClearVbaModules(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#ClearVbaModules"
            Dim document As DevExpress.XtraRichEdit.API.Native.Document = wordProcessor.Document
            document.LoadDocument("Documents\Grimm.docx")
            If document.VbaProject.Modules.Count > 0 Then document.VbaProject.Modules.Clear()
#End Region  ' #ClearVbaModules
        End Sub
    End Module
End Namespace

File: CS/CodeExamples/VbaMacrosActions.cs
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;
using System;
using System.Collections.Generic;

namespace RichEditDocumentServerAPIExample.CodeExamples
{
    public static class VbaMacrosActions
    {
        public static Action<RichEditDocumentServer> ObtainVbaMacrosAction = ObtainVbaMacros;
        public static Action<RichEditDocumentServer> ClearVbaModulesAction = ClearVbaModules;

        static void ObtainVbaMacros(RichEditDocumentServer wordProcessor)
        {
            #region #ObtainVbaMacros
            Document document = wordProcessor.Document;
            document.LoadDocument("Documents\\Grimm.docx");
            if (document.VbaProject.Modules.Count > 0)
                foreach (VbaModule module in document.VbaProject.Modules)
                { document.AppendText("\r\n \u00B7 " + module.Name); }

            #endregion #ObtainVbaMacros
        }

        static void ClearVbaModules(RichEditDocumentServer wordProcessor)
        {
            #region #ClearVbaModules
            Document document = wordProcessor.Document;
            document.LoadDocument("Documents\\Grimm.docx");
            if (document.VbaProject.Modules.Count > 0)
                document.VbaProject.Modules.Clear();
            #endregion #ClearVbaModules
        }

    }
}
